code/API_definitions/network_access_management.yaml

```diff
@@ -0,0 +1,2863 @@
+    - **Device**: A network operator supplied network access device that supports isolated networks.  Multiple devices that support LAN connectivity across the same isolated network (e.g., mesh Wi-Fi, Thread personal area network, etc.) SHOULD be presented as a single device for implementations of this API.
+    - **Network Operator**: Service provider that implements this API and has operational control of supporting devices.
+    - **Service Owner**: Customer of a network operator and resource owner with one or more devices that supports this API across one or more service sites.
+    - **Service Site**: A distinct service owner location, as interpreted by the network operator, with one or more devices.  A service site typically has a one-to-one relationship with the mailing address of the location.  Network operators that support a single identity or interchangeable identities for multiple service sites at the same mailing address (e.g., a residential and business account, an office and common area account, etc.) MUST provide some distinction in the property address (e.g., attn: Home Office, attn: Common Area, etc.).  Multiple service sites, for the same mailing address, but for unrelated service owner identities, do not have this requirement.
+    - **End-User**: The user connecting to an isolated network with a network client.
+    - **Network Client**: End-user equipment connecting to an isolated network (e.g., laptop, phone, IoT device, etc.).
+    - **API Client**: A first- or third-party software system using this API.
+
+    # API Functionality
+    This API allows API clients to manage isolated networks and reboot network access devices. OAuth 2.0 is used for authorizing an identity (resource holder) and scopes are used for role-based authorization.
+
+    ## Authorization
+    Authorization of API clients SHOULD be performed with OAuth 2.0 using the authorization code grant type such that service owners delegate access to their resources exposed by this API to the API client.  PKCE MUST be used when the API client secret cannot be secured (e.g., mobile application or single page application without a secure back-end).  This API MAY also be used with the client credentials grant type, though such a use case only makes sense for first-party network operator API clients or when the service owner operates their own API client (client ID is bound to the service owner's identity).
+
+    ### Scopes
+    OAuth 2.0 scopes are used to protect the service owner's information and privacy.  The general format is `<API_name>-<Resource>-<Permission>`.  A resource may be further broken down into `<Resource_name>-<Field>` or `<Resource_name>-<Role>` for granular permissions. Scopes also control output information as opposed to just authorizing endpoints. Each endpoint definition will clearly state what scopes are needed for access and which are needed to access additional information. Below are some general rules for scopes in this API.

```

Comment:
I think this would be the first API to include fine-grained access to specific fields or role-based access. If needed, it would probably require some attention from CAMARA.

Reply:
Will handle on https://github.com/camaraproject/NetworkAccessManagement/issues/13